006-strimzi-ui.md

```diff
@@ -0,0 +1,187 @@
+
+The below shows how these pieces could integrate.
+
+![Suggested topology](./images/006-topology.png)
+
+Where:
+
+- 'strimzi-api' is a backend server for Kafka data/requests
+- 'Supporting elements' could be other Kafka related deployments such as Mirror Maker, Kafka Connect and Cruise control. These could also be non Kafka related deployments (not deployed as a part of Strimzi), such as LDAP servers, metric stores etc.
+
+### A UI repository, and ways of working in the repository
+
+In addition to the implementation detailed above, I would propose that any UI contributed to Strimzi be managed and developed as follows:
+
+- Be developed in a dedicated repository inside the Strimzi Github organisation
+- Both the UI server and client are contained in this repo - worked as sub modules in a monorepo
+- That design/discussion/implementation and defect issues are kept solely in this repository
+- All development is done in a behavioural manner - focusing on the end user, and the task they are trying to achieve (see [topic list](#topic-list-page) for an example)
+- That the repository operate in a cloud like CI/CD manner; lots of small/little and often deliveries of new function, appropriately sized and gated, allowing for the UI to be shipped at any time
+- That automation, such as [GitHub actions](https://github.com/features/actions) for example, is used as much as possible to support this CI/CD model - eg, automated dependency updates, automated testing run on PR, issue triage etc
+
+### Phased delivery
+
+A UI is more than just what a user sees. It also needs to be supported by backend data, services, and resources. To this end, and in bearing in mind the cloud development model mentioned above, all UI capability (both front and backend) should be provided in a phased manner. [I would suggest initially that a topic listing page is the first deliverable](#topic-list-page), but before that can happen, [prerequisites](#prerequisites) need to be discussed, agreed and implemented.
+
+#### Prerequisites
+
+In order to provide any UI, the whole stack which will support it will need to be discussed and considered. This will include (but not be limited to):
+
+- Setting up the required UI build/packaging for deployment to Strimzi
+- [Integrating the UI with Strimzi itself](#proposed-deployment)
+- A Kafka backend server for Kafka data/requests the UI server can call
+- UI server configurability/capability - including aspects such as;
+  - Transport security
+  - User session, authentication and authorization capabilities
+  - Cluster metadata - listener addresses, number of brokers, etc
+  - Maintenance/tracing support
+- UI client side capabilities
+  - State design
+  - Configuration/capability discovery from the server
+  - Routing and navigation logic
```

Comment:
Any preference on a routing lib? [Reach](https://reach.tech/router/) and [remix](https://remix.run/) both look promising, might be worth considering over traditional react-router. I don't have a strong preference but haven't worked with anything beyond react-router yet so would need to POC a bit before diving into one of the newer ones.

Reply:
Agreed - what would be optimal is something Remix esq - no configuration. Only concern there is the fact it hasnt been released yet, and some of the licence considerations. I am familiar and know react-router, and I think with the GraphQL capabilities the backend could provide (Introspection), we may be able to do something similar.